panel_view.py

```diff
@@ -282,59 +283,45 @@ def draw_on_main_thread(*args, **kwargs):
     sublime.set_timeout(lambda: draw(*args, **kwargs))
 
 
-def get_window_errors(window, errors_by_bid):
-    # type: (sublime.Window, ErrorsByBid) -> ErrorsByBid
+def get_window_errors(window, errors_by_file):
+    # type: (sublime.Window, ErrorsByFile) -> ErrorsByFile
     return {
-        bid: sorted(
+        filename: sorted(
             errors,
             key=lambda e: (e["line"], e["start"], e["end"], e["linter"])
         )
-        for bid, errors in (
-            (bid, errors_by_bid.get(bid))
-            for bid in buffer_ids_per_window(window)
-        )
-        if errors
+        for filename, errors in errors_by_file.items()
+        if errors and errors[0]['window'] == window.id()
```

Comment:
This is probably the only place where you use `window` but if you leave the structure of the function and swap `buffer_ids_per_window` with `filenames_per_window` this should work out ok.

Reply:
Yes of course. Then I would say the filenames the panel should consider per window is the set of `{util.get_filename(v) for v in window.views()}` + all the sets from `affected_filenames_per_filename` for these. This would describe clearly: show all errors from the open files and from their dependencies.